### PR TITLE
Make auto retry codes configurable

### PR DESCRIPTION
**Pull request by pditommaso**
This PR introduces a new config option that allows setting one more exit codes when running a task with Google Batch that should trigger an automatic retry when a task returns an error code and the opinion `google.batch.maxSpotAttempts` is set to a value greater than zero.

By default this option is set to `[50001]`. See the [Google docs](https://cloud.google.com/batch/docs/troubleshooting#vm_preemption_50001) for more details.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -865,6 +871,7 @@ The following settings are available for Google Cloud Batch:
 : :::{versionadded} 23.11.0-edge
   :::
 : Max number of execution attempts of a job interrupted by a Compute Engine spot reclaim event (default: `5`).
+See also `autoRetryExitCodes`.
```
Comment (pditommaso) [inline docs/config.md:874]: @bentsherman How to link each other these settings?

Comment (bentsherman) [reply docs/config.md]: We don't typically do this, but you should be able to put a link target over the config option like `(google-batch-auto-retry-exit-codes)=` and reference it like any other link:
```
: See also: {ref}`google-batch-auto-retry-exit-codes`
```

Comment (pditommaso) [reply docs/config.md]: yeah, it would bye much better having automatic links for each option

File: docs/config.md [hunk 2]
```diff
@@ -845,6 +845,11 @@ Read the {ref}`google-page` page for more information.
 
 The following settings are available for Google Cloud Batch:
 
+`google.batch.autoRetryExitCodes`
+: :::{versionadded} 24.06.0-edge
```
Comment (pditommaso) [inline docs/config.md:849]:
```suggestion
: :::{versionadded} 24.07.0-edge
```

File: docs/config.md (hunk 1 above)
Comment (bentsherman) [reply docs/config.md]: I think Phil and the docs team want to do this at some point. Maybe we can tackle this in the future as we migrate docs

**Pull request merged**